src/core/rewrite/execute.rs

```diff
@@ -258,10 +260,11 @@ pub fn check_out_updated_head(
         }
     };
 
-    let result = git_run_info.run(
+    let result = check_out_commit(
         effects,
+        git_run_info,
         Some(event_tx_id),
-        &[OsStr::new("checkout"), &checkout_target],
+        &checkout_target.to_string_lossy(),
```

Comment:
Ah, I missed this. Branch names are backed by filenames, so we want to use `OsStr`/`OsString` whenever possible to reference them. You can use `impl AsRef<OsStr>` to accept either a `str` or an `OsStr`, rather than potentially mis-encoding the branch name. I fixed this in #192.